relay-sampling/src/lib.rs

```diff
@@ -1072,19 +1082,23 @@ impl SamplingMatch {
         // 3. /transaction is matched with a transaction rule with a factor of 4 and uses as seed abc -> 0.2 * 4 = 0.8 sample rate
         //
         // We can see that we have 3 different samples rates but given the same seed, the random number generated will be the same.
-        let mut seed = event.id.value().map(|id| id.0);
+        let mut seed = event
+            .map_or(Annotated::empty(), |event| event.id.clone())
+            .value()
+            .map(|id| id.0);
         let mut accumulated_factors = 1.0;
 
+        let event_ty = event.and_then(|event| event.ty.value());
         for rule in rules {
             let matches = match rule.ty {
                 RuleType::Trace => match dsc {
                     Some(dsc) => rule.condition.matches(dsc, ip_addr),
                     _ => false,
                 },
-                RuleType::Transaction => match event.ty.0 {
+                RuleType::Transaction => event.map_or(false, |event| match event_ty {
```

Comment:
Do you need `event_ty` somewhere else? If not, this should work:

```suggestion
                RuleType::Transaction => event.map_or(false, |event| match event.py {
```

Reply:
Overlooked this, thanks for the heads up!